### PR TITLE
Issue #230 Changed genesis initialization in Shell

### PR DESCRIPTION
**Pull request by batconjurer**
Changed the genesis initialization in the shell to occur in `Shell::init_chain`. Also fixed some linting

### CODE REVIEW

File: apps/src/lib/node/ledger/mod.rs [hunk 1]
```diff
@@ -296,6 +208,88 @@ impl Shell {
                 current_chain_id, chain_id
             )));
         }
+        if self.last_state().is_none() {
```
Comment (tzemanovic) [inline apps/src/lib/node/ledger/mod.rs:211]: We could either skip this check or make it a fatal error as this shouldn't be called if there is a state

**Review by tzemanovic — APPROVED** [review]
besides the minor comment, LGTM, thanks!

**Comment by batconjurer** [comment]
General question, is the shell intended to be run only by ABCI? If so, we can remove the public visibility on the internal methods and just expose run. Then we could just rely on the correctness of the ABCI and don't need these checks.

**Comment by tzemanovic** [comment]
> General question, is the shell intended to be run only by ABCI? If so, we can remove the public visibility on the internal methods and just expose run. Then we could just rely on the correctness of the ABCI and don't need these checks.

Yup, that sounds good!

**Review by tzemanovic — APPROVED** [review]

**Pull request merged**